airflow/contrib/hooks/gcs_hook.py

```diff
@@ -172,8 +172,9 @@ def download(self, bucket_name, object_name, filename=None):
         if filename:
             blob.download_to_filename(filename)
             self.log.info('File downloaded to %s', filename)
-
-        return blob.download_as_string()
+            return filename
```

Comment:
I think that the value of this result should be consistent. In both cases it should be the content/stream of the file.

Reply:
@mik-laj that is breaking the whole thing as when the blob is huge it eats up all the memory.

Reply:
Can you present your assumptions in the docstring?  I think it's worth describing this behavior and the reason for this behavior, not another behavior.

Reply:
@mik-laj I've updated the docstring

Reply:
@mik-laj so is this a yes? :) or should we wait for another review?

Reply:
I'm on vacation untill Friday. 🇫🇷

Reply:
Enjoy! 🥖

Reply:
Information about this change should be propagated to GCS download operator:
https://github.com/apache/airflow/blob/ec7c67ff95e012d5609a56d358fc6c29d01d9ff6/airflow/contrib/operators/gcs_download_operator.py#L28-L41

Reply:
@mik-laj / @nuclearpinguin done